### PR TITLE
Several bug fixes

### PR DESCRIPTION
**Pull request by Baezon**
1. Prune textures with no polygons, these are useless

2. Use the first scene if theres no default scene in gltf

3. Extend the existing texture map when doing multiple merge textures operations

### CODE REVIEW

File: pof/src/parse.rs [hunk 1]
```diff
@@ -5,12 +5,46 @@ use dae_parser as dae;
...
+        for subobj in self.sub_objects.iter_mut() {
+            for (_, poly) in subobj.bsp_data.collision_tree.leaves_mut() {
+                poly.texture = tex_map[&poly.texture];
+            }
+        }
+
+        // and finally remove the unused textures
+        let mut used_tex_ids = used_tex_ids.iter().collect::<Vec<_>>();
+        used_tex_ids.sort();
+        let new_textures = used_tex_ids.iter().map(|id| self.textures[id.0 as usize].clone()).collect();
```
Comment (digama0) [inline pof/src/parse.rs:43]: You could build this in step 2:
```rust
        let mut tex_map = HashMap::new();
        let mut new_textures = vec![];
        for (i, tex) in self.textures.iter().enumerate() {
            if used_tex_ids.contains(&TextureId(i as u32)) {
                new_textures.push(tex);
                tex_map.insert(TextureId(i as u32), TextureId(tex_map.len() as u32));
            }
        }
```
or for extra fancy, skip the temp:
```rust
        let mut tex_map = HashMap::new();
        for (i, tex) in std::mem::take(&mut self.textures).into_iter().enumerate() {
            if used_tex_ids.contains(&TextureId(i as u32)) {
                self.textures.push(tex);
                tex_map.insert(TextureId(i as u32), TextureId(tex_map.len() as u32));
            }
        }
```

File: pof/src/parse.rs [hunk 2]
```diff
@@ -1642,7 +1679,14 @@ pub fn parse_gltf(path: std::path::PathBuf) -> Model {
         })
         .collect();
 
-    GltfContext { buffers }.parse_top_level_nodes(&mut model, gltf.default_scene().expect("Default scene in gltf file not set!").nodes());
+    let scene = gltf
+        .default_scene()
+        .unwrap_or(gltf.scenes().next().expect("no scene found in gltf file!"));
```
Comment (digama0) [inline pof/src/parse.rs:1684]:
```suggestion
        .unwrap_or_else(|| gltf.scenes().next().expect("no scene found in gltf file!"));
```

Comment (Baezon) [reply pof/src/parse.rs:1684]: Why?

Comment (digama0) [reply pof/src/parse.rs:1684]: Well the function calls there are unnecessary, but in particular it seems bad to run something that can panic. I believe there is a clippy warning for this: always use delayed evaluation in this situation unless the default is just a plain value (e.g. a literal or a struct constructor expression of such)

**Pull request merged**